nixos/modules/services/system/nix-daemon.nix

```diff
@@ -224,13 +224,17 @@ in
 
       unitConfig.RequiresMountsFor = "/nix/store";
 
-      serviceConfig = {
-        CPUSchedulingPolicy = cfg.daemonCPUSchedPolicy;
-        IOSchedulingClass = cfg.daemonIOSchedClass;
-        IOSchedulingPriority = cfg.daemonIOSchedPriority;
-        LimitNOFILE = 1048576;
-        Delegate = "yes";
-      };
+      serviceConfig =
+        {
+          CPUSchedulingPolicy = cfg.daemonCPUSchedPolicy;
+          IOSchedulingClass = cfg.daemonIOSchedClass;
+          IOSchedulingPriority = cfg.daemonIOSchedPriority;
+          LimitNOFILE = 1048576;
+          Delegate = "yes";
+        }
+        // lib.optionalAttrs (nixPackage.pname == "lix" && lib.versionAtLeast nixPackage.version "2.94") {
+          DelegateSubgroup = "supervisor";
```

Comment:
IMHO, this change can be applied to Nix or anything without any problem, including if Nix uses the cgroups feature.

cc @r-vdp